src/diffusers/pipelines/amused/pipeline_amused.py

```diff
@@ -0,0 +1,307 @@
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import torch
+from transformers import CLIPTextModelWithProjection, CLIPTokenizer
+
+from ...image_processor import VaeImageProcessor
+from ...models import UVit2DModel, VQModel
+from ...schedulers import AmusedScheduler
+from ...utils import replace_example_docstring
+from ..pipeline_utils import DiffusionPipeline, ImagePipelineOutput
+
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import torch
+        >>> from diffusers import AmusedPipeline
+
+        >>> pipe = AmusedPipeline.from_pretrained(
+        ...     "huggingface/amused-512", variant="fp16", torch_dtype=torch.float16
+        ... )
+        >>> pipe.vqvae.to(torch.float32)  # vqvae is producing nans in fp16
```

Comment:
This should be handled inside the pipeline like we do in sdxl pipeline, we could add a argument to vq model called 
`force_upcast`  if it's `True` the model will be upcasted inside the pipeline.

Reply:
Ah ok, didn't know we did that. Will do

Reply:
done and added fp16 integration tests!